### PR TITLE
execgen: don't depend on sqlbase

### PR DESCRIPTION
**Pull request by jordanlewis**
Previously, execgen accidentally had a dep chain that took it to
sqlbase. Sqlbase is a giant dep that changes all the time, which caused
way too many execgen recompiles. This should improve things a bit.

cc @irfansharif 

Release note: None

### CODE REVIEW

**Review by yuzefovich — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45762#-:-M1g5JA_971xPdmlgS26:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto)

**Comment by irfansharif** [comment]
LGTM. I actually tried this exact diff earlier, though I was trying to wean off the protobuf depedency here: https://github.com/cockroachdb/cockroach/blob/d6c3a5e8182f424b58170134de16d7716a9e5526/Makefile#L1664 

The thing is, `execgen` still transitively depends on `pkg/util` (because literally everything does), which in turn depend on protobufs. Whenever _those_ get regenerated, `execgen` would as well, so I gave up trying that approach and instead opted to not even bother regenerating `execgen` unless explicitly asked for (which seems like a fine tradeoff to me).

**Comment by jordanlewis** [comment]
Protobufs don't change too much though. Not nearly as much as sqlbase and its transitive deps. I think it's gonna be hard to wean off of protobufs.

What do you mean not regenerating execgen unless specifically asked for?

**Comment by irfansharif** [comment]
The dependency chart attached below, if you're curious. Removing `sqlbase`, AFAICT, didn't help much.

godepgraph

**Comment by irfansharif** [comment]
> What do you mean not regenerating execgen unless specifically asked for?

The hack I introduced here. TL;DR: Don't regenerate `bin/execgen` if it's already present.
https://github.com/cockroachdb/cockroach/blob/feb2545bc726a41bdf03f6b63c364addee2f2de9/Makefile#L1681-L1699

**Comment by jordanlewis** [comment]
But that hack will probably have a chance of generating incorrect code from time to time. I don't like it.

**Comment by irfansharif** [comment]
I'll try measuring how much time is spent actually regenerating code vs. re-compiling it tomorrow, after #45737 lands. My only aim with #45611 was to see how fast I could get the `make clean{,short}; make buildshort` cycle (but like you mentioned there, I/everyone should probably simply not use `make clean{,short}` at all). 

That aside, does the behavior of `bin/execgen` really change that often? Or for very many people? I was (selfishly) hoping that given we'd _still_ be doing the right thing when `*_tmpl.go` or code under `pkg/sql/colexec/execgen/cmd/execgen` changed, it's "safe enough" to just ignore transitive dependencies beyond that. And for when maintainers _don't_ want to ignore those dependencies (rare case), they'd just manually rebuild `bin/execgen`.

**Comment by jordanlewis** [comment]
> That aside, does the behavior of bin/execgen really change that often? Or for very many people? I was (selfishly) hoping that given we'd still be doing the right thing when *_tmpl.go or code under pkg/sql/colexec/execgen/cmd/execgen changed, it's "safe enough" to just ignore transitive dependencies beyond that. And for when maintainers don't want to ignore those dependencies (rare case), they'd just manually rebuild bin/execgen.

I don't think this is really true unfortunately. And if we even had a single instance of this, then all of a sudden things would behave weirdly unless we told people to `make clean` more. And right now, you don't ever have to `make clean` because everything is (or should be) getting compiled again if it has to be.

**Pull request merged**
